executor/read/immediate_executor.rs

```diff
@@ -242,6 +242,7 @@ impl IntersectionExecutor {
                     return Ok(true);
                 } else {
                     self.iterators.clear();
+                    self.cartesian_iterator.clear();
                     let _ = self.input.as_mut().unwrap().next().unwrap().map_err(|err| err.clone());
```

Comment:
We can't reuse the iterators created from the previous input row, as they would iterate over different tuples, so we reset the cartesian entirely.